### PR TITLE
feat: add delete TM functionality to CLI and REST API

### PR DESCRIPTION
**Pull request by alexbrdn**
- feat: add delete TM functionality to CLI and REST API

### CODE REVIEW

File: internal/remotes/fs.go [hunk 1]
```diff
@@ -83,6 +84,27 @@ func (f *FileRemote) Push(id model.TMID, raw []byte) error {
...
+	match, _ := f.getExistingID(id)
+	if match != idMatchFull {
+		return ErrTmNotFound
+	}
+	fullFilename, _, _ := f.filenames(id)
+	err = os.Remove(fullFilename)
+	if os.IsNotExist(err) {
+		return ErrTmNotFound
+	}
+	return err
```
Comment (enricoschuetz) [inline internal/remotes/fs.go:105]: Would it be possible to remove also parent folder structure up to the root, if it as the last TM ?
Or would that be too error-prone, e.g. checking if also TM's exists in sub folders (opt-tree / opt-path) ?

File: cmd/delete.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	ValidArgsFunction: nil, // just to make explicit in code that a completion function is not wanted here. Don't want to do anything to make deletion easier
+}
+
+func init() {
+	RootCmd.AddCommand(deleteCmd)
+	deleteCmd.Flags().StringP("remote", "r", "", "name of the remote to delete from")
+	_ = deleteCmd.RegisterFlagCompletionFunc("remote", completion.CompleteRemoteNames)
+	deleteCmd.Flags().StringP("directory", "d", "", "TM repository directory")
+	_ = deleteCmd.MarkFlagDirname("directory")
+	deleteCmd.Flags().StringP("force", "", "", "force the deletion") // intentionally a string flag, not boolean, so that the user has to make that much extra effort to type
```
Comment (enricoschuetz) [inline cmd/delete.go:30]: If we have no shorthand,, we should use the short version without `P` -> `deleteCmd.Flags().String(...`
In the other commands we should correct that than too...

File: internal/remotes/fs.go (hunk 1 above)
Comment (alexbrdn) [reply internal/remotes/fs.go]: added removing empty parent folders up to root. It seems to me, at this late hour on Friday evening, that there shouldn't be any problems with that :)

File: cmd/delete.go (hunk 2 above)
Comment (alexbrdn) [reply cmd/delete.go]: good point. changed to `String()`

**Pull request merged**